sample-apps/javascript-sample-app/server.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS'" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+'use strict'
+
+const sdk = require("./common");
+const Worker = require("worker_threads");
+const http = require('http');
+const AWS = require('aws-sdk');
+const fetch = require ("node-fetch");
+
+// tracer
+const api = require('@opentelemetry/api'); 
+const common_span_attributes = { signal: 'trace', language: 'javascript' };
+
+// config
+const create_cfg = require('./config');
+const cfg = create_cfg.create_config('./config.yaml');
+
+
+// initialise sdk (metric and trace provider) and start server and a separate thread for random-metrics generation.
+sdk.nodeSDKBuilder()
+    .then(() => {
+    startServer();
```

Comment:
why random metrics cannot be started from here?

Reply:
Is there a reason why we need an worker for the random metrics? `startServer` is non blocking, so it should be fine to use a single sdk for both requests and metrics.

Reply:
I have initialized `nodeSDK` both from `main` and `worker thread`. 
Just initializing the `nodeSDK` in `main` and not in `workerThread` gives `noop meter` in `worker thread` - even when the `worker thread` is started after the initialization of `nodeSDK` in `main`. 
So, the only solution I found was to initialize `nodeSDK` in both.

Reply:
> why random metrics cannot be started from here?

`RandomMetrics` are already starting at this point? could you please tell me more.

Reply:
> Is there a reason why we need an worker for the random metrics? `startServer` is non blocking, so it should be fine to use a single sdk for both requests and metrics.

As per my understanding we do need some separate thread/process for the `randomMetrics` generation. The design of 'workerThread' was selected by Nick during the initial phase. I kept it same.

Reply:
I think you can change Nick`s design :) . You can run both from the same thread using the async mechanisms of js, no?
Eg.:
```
sdk.nodeSDKBuilder()
    .then(() => {
    startServer()
    setInterval(() => {
        updateTimeAlive();
        updateThreadsActive();
        updateCpuUsageMetric();
        updateSizeMetric();
    }, cfg.TimeInterval * 1000);
```

Node is asynchronous first, and not multi threaded. Threre cannot be shared variables between the main thread and a worker. they can only communicate through messages, and that is why you were getting No Op instruments before.

Reply:
https://blog.logrocket.com/node-js-multithreading-worker-threads-why-they-matter/

Reply:
Thanks. Updated.